spec/tasks/combine_schema_spec.rb

```diff
@@ -168,4 +169,48 @@
       expect(actual.schema).to eq(expected.schema)
     end
   end
+
+  context "message queue schema" do
+    let(:output_filename) { File.join("dist", "message_queue.json") }
+    let(:details) { build_schema('details.json', properties: build_string_properties('detail')) }
+    let(:publisher_schema_dir) { project_root.join("formats", format_name, "publisher") }
+
+    around do |example|
+      format_root = ENV["FORMAT_ROOT"]
+      FileUtils.mv("dist/message_queue.json", "dist/message_queue.json.bkp")
+
+      Dir.mktmpdir do |dir|
+        test_root = ENV["FORMAT_ROOT"] = "#{dir}"
+
+        ["format_foo", "format_bar"].each do |format|
+          FileUtils.mkdir_p("#{test_root}/#{format}/publisher")
+        end
+
+        File.write("#{test_root}/message_queue_base.json", "{}")
+        File.write("#{test_root}/definitions.json", "{}")
+        File.write("#{test_root}/base_links.json", "{}")
+
+        File.write("#{test_root}/format_foo/publisher/details.json", "{}")
+
+        example.run
+      end
+
+      FileUtils.mv("dist/message_queue.json.bkp", "dist/message_queue.json")
+      ENV["FORMAT_ROOT"] = format_root
+    end
+
+    specify "the frontend schema file contains a frontend schema" do
```

Comment:
I'm finding it really hard to tell what this is actually testing. Why frontend schemas? Is the point that we're asserting that this is called for format_foo but not format_bar? It would be useful to be more explicit.